src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -299,8 +304,18 @@ private void processData(InputData data) {
 				if (state != null) {
 					try {
 						State stateModel = gson.fromJson(state, State.class);
-						dataModel.latestState
-							.put(record.serialNumber, stateModel);
+						List<State> latestStatesList = dataModel.latestStates
+							.computeIfAbsent(
+								record.serialNumber,
+								k -> Collections
+									.synchronizedList(new LinkedList<>())
```

Comment:
Why is this `synchronizedList` ??

Reply:
My intention is to make the internal structure thread-safe as well. Though a `ConcurrentMap` is already thread-safe

Reply:
Is it actually needed? It will hurt performance (always locks), whereas `ConcurrentMap` is less likely to introduce overhead.

Reply:
If we want it to be thread-safe. It is needed. Because the List here as a value is not guaranteed to be thread-safe.

Reply:
This is the only place actually modifying the list, right? In that case you don't need thread safety. In RRM algorithms, the whole `DataModel` is copied.